tests/foreman/rhai/test_rhai_client.py

```diff
@@ -69,21 +73,106 @@ def setUpClass(cls):  # noqa
+        """Verify that '--test-connection' option for insights-client rpm tests
+        the connection with the satellite server connection with satellite
+        server
 
         :id: 167758c9-cbfa-4a81-9a11-27f88aaf9118
 
-        :expectedresults: 'redhat-access-insights --test-connection' should
+        :expectedresults: 'insights-client --test-connection' should
             return zero on a successfully registered machine to RHAI service
         """
         with VirtualMachine(distro=DISTRO_RHEL6) as vm:
             vm.configure_rhai_client(self.ak_name, self.org_label,
                                      DISTRO_RHEL6)
             test_connection = vm.run(
-                'redhat-access-insights --test-connection')
+                'insights-client --test-connection')
             self.logger.info('Return code for --test-connection {0}'.format(
                 test_connection.return_code))
             self.assertEqual(test_connection.return_code, 0,
                              '--test-connection check was not successful')
+
+    def test_provision(self):
+        """A new host appears in Red Hat Access Insights inventory
+
+        :id: 0bf2bccf-0973-49fa-923f-7409fba86f85
+
+        :expectedresults: provisioned host is in the inventory
+        """
+        with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+            vm.configure_rhai_client(
+                activation_key=self.ak_name,
+                org=self.org_label,
+                rhel_distro=DISTRO_RHEL7)
+            self.assertIn(
+                'Client: 3.',
+                vm.run('insights-client --version').stdout[0],
+            )
+            with Session(self) as session:
+                set_context(session, org=self.org_label)
+                vm_info = session.hosts.get_host_properties(
+                    vm.hostname,
+                    ['Status', 'Operating system', 'Subscription'])
+                self.assertIn(vm_info['Status'], ['Warning', 'OK'])
+                self.assertEqual(vm_info['Subscription'], 'Fully entitled')
+                self.assertTrue(
+                    session.rhai_inventory.search(vm.hostname).is_displayed())
+
+    def test_numeric_group(self):
+        """Check the rule appears when provoked and disappears on Satellite
+        once applied
+
+        :id: 4562e0f9-fff1-4cc7-b7e7-e4779662b3e1
+
+        :expectedresults: rule no more appears on Rules page on portal
+        """
+        message = 'group names are numeric'
+        with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+            vm.configure_rhai_client(
+                activation_key=self.ak_name,
+                org=self.org_label,
+                rhel_distro=DISTRO_RHEL7)
+
+            with Session(self) as session:
+                set_context(session, org=self.org_label)
+
+                session.rhai_inventory.search_and_click(vm.hostname)
+                self.assertIsNotNone(
+                    session.nav.wait_until_element_exists(
```

Comment:
`wait_until_element` is widely used method, because it verifies not only existence, but visibility